Lazy loading for the word list in the popup

villevpe opened the pull request:
Implementation for #4 with scroll position based lazy loading. It works by rendering new word items to the list when user scrolls to near the bottom of the popup (if there are still items to be rendered). 

I also refactored the rendering of the list a little bit so it is easier to read. Hope you don't mind the added eslint rules, I find them necessary :)

Commit:
Implement scroll position based lazy loading

Commit:
Easen up eslint rules a bit

shreyas-a commented:
@villevpe for some reason meanings are not fetched after 10. I guess scroll logic needs a fix.
Really appreciate your efforts towards this, and sure you can have those eslint rules.

villevpe commented:
Thats odd, I'll look into it

Commit:
Rebase delete-action, fix scrolltop issue

villevpe commented:
Rebased with master & fixed the issue. Turns out that with some versions of chrome you should use `document.body.scrollTop` to get the scroll position while others work with `document.documentElement.scrollTop`. Added fallbacks for both cases (+`window.pageYOffset` to play it safe)  :)
